src/server/systems/ServerGameController.cpp

```diff
@@ -83,9 +90,19 @@ void ServerGameController::postUpdate(sf::Int32 update_time) {
 void ServerGameController::setNetworkState() {
     GameState game_state = {
         .object = m_gameObjectController->getGameStateObject(),
+        .event = m_gameStateEvent,
         .core = m_gameStateCore,
     };
     m_networkingServer->setState(game_state);
+    m_gameStateEvent.collisions.clear();
```

Comment:
is there no lock needed here?

Reply:
no, the lock happens in 
`m_networkingServer->setState(game_state);`

m_gameStateEvent is just a local variable only used by this class